app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -55,6 +55,22 @@ open class ProfileViewModel @Inject constructor(private val repository: Profiles
         onFailure = {})
   }
 
+  fun addLikedActivity(userId: String, activityId: String) {
+    repository.addLikedActivity(
+        userId = userId,
+        activityId = activityId,
+        onSuccess = { fetchUserData(userId) },
+        onFailure = {})
+  }
+
+  fun removeLikedActivity(userId: String, activityId: String) {
+    repository.removeLikedActivity(
+        userId = userId,
+        activityId = activityId,
+        onSuccess = { fetchUserData(userId) },
+        onFailure = {})
```

Comment:
The implementation for adding and removing liked activities within the ProfileViewModel appears clean and functional. The structure of these functions shows a clear handling of asynchronous operations which is a good practice in maintaining responsiveness in the user interface.